pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -370,7 +378,15 @@ func (s *Service) LaunchTemplateNeedsUpdate(scope *scope.MachinePoolScope, incom
 
 	incomingIDs := make([]string, len(incoming.AdditionalSecurityGroups))
 	for i, ref := range incoming.AdditionalSecurityGroups {
-		incomingIDs[i] = aws.StringValue(ref.ID)
+		if ref.ID != nil {
+			incomingIDs[i] = aws.StringValue(ref.ID)
+		} else if ref.Filters != nil {
+			group, err := s.GetFilteredSecurityGroupID(ref)
+			if err != nil {
+				return false, err
+			}
+			incomingIDs[i] = group
+		}
 	}
```

Comment:
This logic is copy/pasted 3 times in this file, also I see the exact same logic in `awsmachine_security_groups.go`.

Is there a way to make a method out of this?

Reply:
I will refactor this code.

Reply:
Done